### DISCUSSION TITLE
Make the accepted file type be dependent on the select option

### DISCUSSION THREAD

**Discussion by KiyaTilahun** [filamentphp/filament#14097] [Help]
### Package

Panel builder

### Package Version

v 3.0.0

### How can we help you?

              Select::make('media_type_id')
                    ->label('Media Type')->reactive()
                    ->options(fn() => MediaType::pluck('name', 'id'))->helperText('File/Folder Type')->required()->label('Folder'),
           
                FileUpload::make('file_path')
                    ->multiple()
                    ->reorderable()->maxSize( 1048576)
                    ->live()->helperText('Choose File Depending on the file type you choosed | multiple file allowed')              

I want the fileupload acceptedfiletypes to use the select option value which are   ['name' => 'Image'],
            ['name' => 'Video'],
            ['name' => 'Document'],
            ['name' => 'Audio'],
            ['name' => 'Mixed'],  that are in the mediatype table
I tried afterstateupdated on the select option  but it does not update it

**Comment by leandrocfe** [comment]
```php
Select::make('types')
    ->live()
    ->options([
        'audio' => 'Audio',
        'image' => 'Image',
        'video' => 'Video',
    ]),
FileUpload::make('document')
    ->acceptedFileTypes(fn (Get $get) => match ($get('types')) {
        default => null,
        'video' => ['video/mpeg', 'video/mp4'],
        'image' => ['image/jpeg', 'image/png'],
        'audio' => ['audio/mpeg']
    })
```

**Reply by KiyaTilahun** [reply]
it still does not work i am trying to make a folder management for a CMS  project .   i tried different methods but no solution.  Does it work for you . it does not update the accepted file type .

**Reply by michelbarutot** [reply]
Hey! I had the same issue, I solved it using the `rules` method to validate the mimes:

```
->rules(fn (Get $get) => [
    'mimetypes:'.implode(',', SiteMediaTypesEnum::getMimeTypes($get('type'))),
])
```

Hope it helps!
